tests/test_linear_feedback_controller.cpp

```diff
@@ -0,0 +1,378 @@
+    Expectations out;
+    out.pd = ExpectedPDControlFrom(gains, refs, sensor.joint_state.position,
+                                   sensor.joint_state.velocity);
+    out.lf = ExpectedLFControlFrom(model, sensor, control);
+    return out;
+  }
+};
+
+struct LinearFeedbackControllerTest
+    : public ::testing::TestWithParam<ControllerParameters> {};
+
+TEST(LinearFeedbackControllerTest, Ctor) {
+  EXPECT_NO_THROW({ auto ctrl = LinearFeedbackController{}; });
+}
+
+TEST(LinearFeedbackControllerTest, DISABLED_LoadEmptyParams) {
+  auto ctrl = LinearFeedbackController{};
+  EXPECT_FALSE(ctrl.load(ControllerParameters{}));
+  EXPECT_EQ(ctrl.get_robot_model(), nullptr);
+}
+
+TEST_P(LinearFeedbackControllerTest, DISABLED_LoadNoURDF) {
+  auto ctrl = LinearFeedbackController{};
+  auto no_urdf_param = GetParam();
+  no_urdf_param.urdf.clear();
+
+  EXPECT_FALSE(ctrl.load(no_urdf_param));
+  EXPECT_EQ(ctrl.get_robot_model(), nullptr);
+}
+
+TEST_P(LinearFeedbackControllerTest, DISABLED_LoadSizeMismatch) {
+  auto ctrl = LinearFeedbackController{};
+  auto good_params = GetParam();
+
+  {
+    auto p_gains_too_big = good_params;
+    p_gains_too_big.p_gains.push_back(3.141592);
+
+    EXPECT_FALSE(ctrl.load(p_gains_too_big));
+    EXPECT_EQ(ctrl.get_robot_model(), nullptr);
+  }
+
+  {
+    auto p_gains_smaller = good_params;
+    p_gains_smaller.p_gains.pop_back();
+
+    EXPECT_FALSE(ctrl.load(p_gains_smaller));
+    EXPECT_EQ(ctrl.get_robot_model(), nullptr);
+  }
+
+  {
+    auto d_gains_too_big = good_params;
+    d_gains_too_big.d_gains.push_back(3.141592);
+
+    EXPECT_FALSE(ctrl.load(d_gains_too_big));
+    EXPECT_EQ(ctrl.get_robot_model(), nullptr);
+  }
+
+  {
+    auto d_gains_smaller = good_params;
+    d_gains_smaller.d_gains.pop_back();
+
+    EXPECT_FALSE(ctrl.load(d_gains_smaller));
+    EXPECT_EQ(ctrl.get_robot_model(), nullptr);
+  }
+
+  {
+    auto moving_joint_cleared = good_params;
+    moving_joint_cleared.moving_joint_names.clear();
+
+    EXPECT_FALSE(ctrl.load(moving_joint_cleared));
+    EXPECT_EQ(ctrl.get_robot_model(), nullptr);
+  }
+}
+
+TEST_P(LinearFeedbackControllerTest, DISABLED_LoadNegativeDuration) {
+  auto ctrl = LinearFeedbackController{};
+
+  auto negative_duration_params = GetParam();
+  negative_duration_params.pd_to_lf_transition_duration =
+      -negative_duration_params.pd_to_lf_transition_duration;
+
+  EXPECT_FALSE(ctrl.load(negative_duration_params));
+  EXPECT_EQ(ctrl.get_robot_model(), nullptr);
+}
+
+TEST_P(LinearFeedbackControllerTest, Load) {
+  auto ctrl = LinearFeedbackController{};
+  EXPECT_TRUE(ctrl.load(GetParam()));
+  EXPECT_NE(ctrl.get_robot_model(), nullptr);
+}
+
+TEST_P(LinearFeedbackControllerTest, DISABLED_SetInitialStateEmpty) {
+  auto ctrl = LinearFeedbackController{};
+  ASSERT_TRUE(ctrl.load(GetParam()));
+  EXPECT_FALSE(ctrl.set_initial_state({}, {}));
+}
+
+TEST_P(LinearFeedbackControllerTest, DISABLED_SetInitialStateSizeMismatch) {
```

Comment:
Is this one also doable with `TemporaryMutate`?

Reply:
for 4 scopes?

Reply:
As already discussed. We can skip it